KeyVault.Acmebot/Providers/GoDaddyProvider.cs

```diff
@@ -81,13 +81,31 @@ public GoDaddyClient(string apiKey, string apiSecret)
 
         public async Task<IReadOnlyList<ZoneDomain>> ListZonesAsync()
         {
-            var response = await _httpClient.GetAsync("v1/domains?statuses=ACTIVE&includes=nameServers");
+            var limit = 100;
+            var marker = string.Empty;
+            var allActiveDomains = new List<ZoneDomain>();
 
-            response.EnsureSuccessStatusCode();
+            do
+            {
+                var response = await _httpClient.GetAsync($"v1/domains?statuses=ACTIVE&includes=nameServers&limit={limit}{marker}");
+
+                response.EnsureSuccessStatusCode();
+                marker = string.Empty;
+
+                var domains = await response.Content.ReadAsAsync<ZoneDomain[]>();
+
+                allActiveDomains.AddRange(domains);
 
-            var domains = await response.Content.ReadAsAsync<ZoneDomain[]>();
+                if (domains.Length == limit)
+                {
+                    if (domains.Last().Domain is not null)
+                    {
+                        marker = $"&marker={domains.Last().Domain}";
+                    }
+                }
+            } while (marker != string.Empty);
```

Comment:
I'm not familiar with the GoDaddy API, but could this be accomplished with code as simple as this?

```csharp
            var limit = 100;
            var marker = "";
            var allActiveDomains = new List<ZoneDomain>();

            while (true)
            {
                var response = await _httpClient.GetAsync($"v1/domains?statuses=ACTIVE&includes=nameServers&limit={limit}{marker}");

                response.EnsureSuccessStatusCode();

                var domains = await response.Content.ReadAsAsync<ZoneDomain[]>();

                if (domains.Length == 0)
                {
                    break;
                }

                allActiveDomains.AddRange(domains);

                marker = $"&marker={domains[^1].Domain}";
            }
```

Reply:
Looks good to me!